packages/happn-cluster/lib/services/orchestrator.js

```diff
@@ -1,504 +1,402 @@
-    _this = this;
-  var peerCount;
-  var errors = [];
-  var reports = [];
-  var unstableCount = 0;
-
-  Object.keys(this.members).forEach(function (memberId) {
-    var error,
-      missing,
-      member = _this.members[memberId];
-
-    // skip existing peers, already stable
-    if (member.name && _this.peers[member.name]) return;
-
-    error = false;
-    missing = [];
-
-    Object.keys(_this.stableExpectationMask).forEach(function (key) {
-      var expectedValue = _this.stableExpectationMask[key];
-
-      if (key === 'error' && member.error) {
-        errors.push(member.error);
-        error = true;
-        return;
-      }
-
-      if (expectedValue === 'defined') {
-        if (!member[key]) missing.push(key);
-        return;
-      }
-
-      if (expectedValue !== member[key]) return missing.push(key);
+  defaults(config) {
+    config = config || {};
+    config = _.defaultsDeep({}, config, {
+      serviceName: 'happn-cluster-node',
+      deployment: 'Test-Deploy',
+      clusterName: 'happn-cluster',
+      timing: {
+        memberRefresh: 5e3,
+        keepAlive: 5e3,
+        keepAliveThreshold: 6e3,
+        healthReport: 10e3,
+      },
     });
+    if (config.stabiliseTimeout) config.timing.stabiliseTimeout = config.stabiliseTimeout;
+    config.replicate = config.replicate || ['*'];
+    config.replicate.push('/__REPLICATE');
+    config.replicate = this.__reducePaths(config.replicate);
+    if (config.cluster && Object.keys(config.cluster).length) return config;
+    config.cluster = {};
+    config.cluster[config.serviceName] = config.minimumPeers || 1;
+    return config;
+  }
 
-    if (!error && missing.length === 0) return _this.__addPeer(member);
-
-    unstableCount++;
-
-    if (report) {
-      reports.push(
-        util.format('member %s awaiting %s', member.name || member.memberId, missing.join(', '))
-      );
+  async start() {
+    this.endpoint = this.announceHost
+      ? this.announceHost + ':' + this.happn.config.port
+      : this.ip + ':' + this.happn.config.port;
+    this.loginConfig = {
+      // used to login to remote cluster members as a cluster peer
+      info: {
+        name: this.happn.name, // a.k.a. mesh.name
+        clusterName: this.clusterName,
+        serviceName: this.serviceName,
+        endpoint: this.endpoint,
+      },
+    };
+    if (this.secure) {
+      this.adminUser = this.happn.services.security.config.adminUser;
+      this.loginConfig.username = this.adminUser.username;
+      this.loginConfig.password = this.adminUser.password;
     }
-  });
-
-  if (errors.length > 0) {
-    // each erroring member already logged it's error
-    clearInterval(this.stableReportInterval);
-    clearInterval(this.stableTimeout);
-    // can only callback with one error
-    while ((callback = this.stabiliseWaiting.shift()) !== undefined) callback(errors[0]);
-    return;
+    var localLoginConfig = clone(this.loginConfig);
+    delete localLoginConfig.info.clusterName;
```

Comment:
add comment back:
/** 
remove clusterName for localLogin (to self) so that
in happner-cluster our self connection receives replicated events
ie. not filtered out by happn when noCluster is set
**/

Reply:
done